src/workerd/io/hibernation-manager.c++

```diff
@@ -144,12 +155,32 @@ kj::Promise<void> HibernationManagerImpl::readLoop(HibernatableWebSocket& hib) {
   // Like the api::WebSocket readLoop(), but we dispatch different types of events.
   auto& ws = *hib.ws;
   return ws.receive()
-      .then([this, &hib] (kj::WebSocket::Message&& message) mutable -> kj::Promise<void> {
+      .then([this, &hib, &ws] (kj::WebSocket::Message&& message) mutable -> kj::Promise<void> {
     // TODO(now): Within CustomEvent impl run, when we pass a reference to this hibernation manager,
-    // we'll want to assign the hib manager ref to the Worker::Actorso that JS can access it elsewhere.
-    if (hib.activeWebSocket == nullptr) {
+    // we'll want to assign the hib manager ref to the Worker::Actor so that JS can access it elsewhere.
+
+    auto unhibernate = true;
+    KJ_IF_MAYBE (req, autoResponseRequest) {
+      // If we have an autoResponseRequest set, we must have
+      // autoResponseResponse also.
+      KJ_ASSERT_NONNULL(autoResponseResponse);
+      KJ_SWITCH_ONEOF(message) {
+        KJ_CASE_ONEOF(text, kj::String) {
+          if (text == kj::str(req)) {
```

Comment:
I think this can be `text == *req` (avoids an allocation), but I'm having some trouble compiling because of a mismatch between our internal repo and workerd commits. Mind opening a branch on our internal repo that points workerd to this this workerd branch?